pkg/build/registry/buildconfig/webhook.go

```diff
@@ -72,9 +73,13 @@ func (w *WebHook) ServeHTTP(writer http.ResponseWriter, req *http.Request, ctx k
 		Revision:    revision,
 		Env:         envvars,
 	}
-	if _, err := w.instantiator.Instantiate(config.Namespace, request); err != nil {
+	newBuild, err := w.instantiator.Instantiate(config.Namespace, request)
+	if err != nil {
 		return errors.NewInternalError(fmt.Errorf("could not generate a build: %v", err))
 	}
+	// Send back the build name so that the client can alert the user.
+	newBuildJson, _ := json.Marshal(newBuild)
```

Comment:
Isn't this an internal version of the object?  If so, then no client will be able to deserialize it.  You'd need to go through a `runtime.Encode` path using a codec.

Also, if a caller is expecting a status object or a build object, writing something other than those things seems wrong (for the reasons @bparees already called out).  Writing a status success (the call actually did succeed right?) that includes a message (intended to be human readable) that explains the situation seems more appropriate.